src/agoradatatools/process.py

```diff
@@ -123,12 +125,22 @@ def process_dataset(
             staging_path=staging_path,
             filename=dataset_name + "." + dataset_obj[dataset_name]["final_format"],
         )
-    else:
+    elif isinstance(df, list):
+        json_path = load.list_to_json(
+            df=df,
+            staging_path=staging_path,
+            filename=dataset_name + "." + dataset_obj[dataset_name]["final_format"],
+        )
+    elif isinstance(df, DataFrame):
         json_path = load.df_to_json(
             df=df,
             staging_path=staging_path,
             filename=dataset_name + "." + dataset_obj[dataset_name]["final_format"],
         )
+    else:
+        raise ADTDataProcessingError(
+            f"Data processing failed for {dataset_name}. Dataframe type is not supported."
+        )
```

Comment:
What do we think about this error handling? Is it necessary? Since we control all of the outputs, maybe we don't need it and can keep using the `else:` for the case where `df` is a `pd.DataFrame`.

Reply:
I'll let Brad or others talk about the necessity of this, my comment is on the message itself.

If I saw this message "in a vacuum" I would ask myself "What WAS the Dataframe type when the exception was raised?"

could that information be added to the exception?

Reply:
Really great point! I updated the error message, let me know if you think I should make any more changes here.

Reply:
Yeah I don't think this is needed because of us being in control of the data structures that come out of the transform functions as you said @beatrizsaldana.